provisioner/config.go

```diff
@@ -1,122 +1,74 @@
 package provisioner
 
 import (
+	"encoding/json"
 	"errors"
 	"fmt"
 	"io/ioutil"
 	"os"
 
-	"github.com/resin-os/resin-provisioner/resin"
 	"github.com/resin-os/resin-provisioner/util"
 )
 
 type Config struct {
-	Uuid                  string `json:"uuid"`
-	ApplicationId         string `json:"applicationId"`
-	ApiKey                string `json:"apiKey"`
-	UserId                string `json:"userId"`
-	UserName              string `json:"username"`
-	DeviceId              int64  `json:"deviceId",omitempty`
-	DeviceType            string `json:"deviceType"`
-	RegisteredAt          int64  `json:"registered_at,omitempty"`
-	AppUpdatePollInterval string `json:"appUpdatePollInterval"`
-	ListenPort            string `json:"listenPort"`
-	VpnPort               string `json:"vpnPort"`
-	ApiEndpoint           string `json:"apiEndpoint"`
-	VpnEndpoint           string `json:"vpnEndpoint"`
-	RegistryEndpoint      string `json:"registryEndpoint"`
-	DeltaEndpoint         string `json:"deltaEndpoint"`
-	PubnubSubscribeKey    string `json:"pubnubSubscribeKey"`
-	PubnubPublishKey      string `json:"pubnubPublishKey"`
-	MixpanelToken         string `json:"mixpanelToken"`
-
-	// See json.go/parseConfig() for more details on what this is for.
-	InitialRaw map[string]interface{} `json:"-"`
```

Comment:
IIRC this was in place to avoid omitting any fields from the original config.json - are you sure this doesn't happen now? 
For example, if the config.json has a "version" or "appUpdatePollInterval" field - doesn't this program remove it now? Cause that would cause a bunch of hard-to-hunt bugs.